obj_gen.cpp

```diff
@@ -398,7 +398,7 @@ data_object* object_generator::get_object(int iter)
         new_size = m_data_size.size_fixed;
     } else if (m_data_size_type == data_size_range) {
         if (m_data_size_pattern && *m_data_size_pattern=='S') {
-            float a = (m_key_index-m_key_min)/(m_key_max-m_key_min);
+            float a = (m_key_index-m_key_min)/(float)(m_key_max-m_key_min);
             new_size = (m_data_size.size_range.size_max-m_data_size.size_range.size_min)*a + m_data_size.size_range.size_min;
```

Comment:
This now triggers a compilation warnings on size mismatch - need to add some casts (both here and in random_range two lines below) - and let's use static_cast<> instead of C style...